h/security/permits/engine.py

```diff
@@ -0,0 +1,69 @@
+from enum import Enum
+
+
+def predicate(requires=None):
+    def decorator(function):
+        function.requires = requires or []
+        return function
+
+    return decorator
```

Comment:
Originally this wrapped all the functions in an object for sorting and all sort of other cleverness, but that's not required any more, so all this really does now is tag the parent predicates and ensures the `requires` attribute is present.

This doesn't need to happen on things which don't have any requirements anymore. It was a nice simplification before, but now permission enums are allowed to be "predicates" it doesn't matter.